flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendLoadingTest.java

```diff
@@ -86,153 +83,49 @@ void testApplicationDefinedHasPrecedence() throws Exception {
     }
 
     // ------------------------------------------------------------------------
-    //  Memory State Backend
+    //  HashMap State Backend
     // ------------------------------------------------------------------------
 
-    /** Validates loading a memory state backend from the cluster configuration. */
+    /** Validates loading a HashMapStateBackend from the cluster configuration. */
     @Test
-    void testLoadMemoryStateBackendNoParameters() throws Exception {
+    void testLoadHashMapStateBackendNoParameters() throws Exception {
         // we configure with the explicit string (rather than
         // AbstractStateBackend#X_STATE_BACKEND_NAME)
         // to guard against config-breaking changes of the name
 
-        final Configuration config1 = new Configuration();
-        config1.setString(backendKey, "jobmanager");
-
-        final Configuration config2 = new Configuration();
-        config2.setString(backendKey, MemoryStateBackendFactory.class.getName());
+        final Configuration config = new Configuration();
+        config.setString(backendKey, HashMapStateBackendFactory.class.getName());
 
-        StateBackend backend1 = StateBackendLoader.loadStateBackendFromConfig(config1, cl, null);
-        StateBackend backend2 = StateBackendLoader.loadStateBackendFromConfig(config2, cl, null);
+        StateBackend backend = StateBackendLoader.loadStateBackendFromConfig(config, cl, null);
 
-        assertThat(backend1).isInstanceOf(MemoryStateBackend.class);
-        assertThat(backend2).isInstanceOf(MemoryStateBackend.class);
+        assertThat(backend).isInstanceOf(HashMapStateBackend.class);
     }
 
     /**
-     * Validates loading a memory state backend with additional parameters from the cluster
+     * Validates loading a hashMap state backend with additional parameters from the cluster
      * configuration.
      */
     @Test
-    void testLoadMemoryStateWithParameters() throws Exception {
-        final String checkpointDir = new Path(TempDirUtils.newFolder(tmp).toURI()).toString();
-        final String savepointDir = new Path(TempDirUtils.newFolder(tmp).toURI()).toString();
-        final Path expectedCheckpointPath = new Path(checkpointDir);
-        final Path expectedSavepointPath = new Path(savepointDir);
-
+    void testLoadHashMapStateBackendWithParameters() throws Exception {
```

Comment:
IIUC, you should remove this Test as it test the Hashmap with parameters, but actually there is no parameter set.